dotnet/src/Microsoft.AutoGen/Core/IAgentRuntime.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// IAgentRuntime.cs
+
+using Google.Protobuf;
+using Microsoft.AutoGen.Contracts;
+namespace Microsoft.AutoGen.Core;
+
+public interface IAgentRuntime
```

Comment:
I strongly believe that we should not be exposing Proto classes in the interface since that is just one possible backing implementation. It's possible to implement this via rest and http servers for example, in which case the proto schema is not strictly relevant.

Reply:
That's where all the Messages (Request/Response) are defined. proto is like data class, it's just a data descriptor/serialization specification, it doesn't bind you to grpc. you can use proto with http and streaming. all of the protos get turned into .NET code and classes at build time - from the dev pod its just a type of Message